voting system and hash function to generate anonymous id updated

Hrusi01 opened the pull request:
**update 👍** 

**Vote functionality**

> added a new vote table schema to maintain records 
> up vote 1
> no vote 0
> downvote -1
> pk-userid+postid ensure a person can't register vote multiple time in a single post  


**END POINTS:**

     Auth required
         post    /posts/{postId}/vote      Handles upvote, downvote, and vote retraction transactions (-1,0,1)

**HASH FUNCTION**

this function returns random looking names that cant be traceback to original user id 

Algorithm flow 
uses hash of seed and secret key(**provided by user**) using that in sha 256 
from adjectives and animal  dictionary of 30 each and number 2 digit  0 to 99 we can generate unique unknown uid

30*30*100 =90,000
unique id per thread 

even if the secret key and thread id know HASH cant be reversed and uid is not stored in data base and only known to the current state of the server 

How the system will know about the person?:

when we will pass user id ,thread id and secret key in that hash it will give same random user name out put per unique thread 
by that system can check which user is this 
db wont store the secret key it is only known to the server

**what function expects:**?

a secret key from front end using by the user that to be passed by post request



@iamanishx #38

Hrusi01 committed:
anonymous hash function

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
updated hash

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
finalHash function

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
old hash  file deleted

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
vote schema added

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
index.ts  updated

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
vote.dto.ts and votes.ts created

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 commented:
@iamanishx #38

iamanishx reviewed (commented):
@Hrusi01 how are you supposed to get total number of votes for a post ? And its if someone upvote and then downvote it totally 0 right .

Hrusi01 commented:
> @Hrusi01 how are you supposed to get total number of votes for a post ? And its if someone upvote and then downvote it totally 0 right .

we can find that in post table vote column  as post.vote 
stores net vote 

if someone down vote or up vote 

final score is always calculated by the change in value relative to the previous vote

if a has vote 0 
if up vote the a=+1
if down vote then a= -currentvote-1 =-2 from total

Hrusi01 commented:
> @Hrusi01 the vote mapping table is just to prevent multiple upvote or downvote from a user so i think we can have downvote and upvote as boolean so a user on upvoting will be upvote as true and on downvoting downvote as true it will better to maintain

i think it mean vote table with 3 column  pk, 2 columns upvote(default o  yes if 1 ), down vote(default 0   yes if 1 ) 
here  we need to scan 2 column to get 3 variable 10 or 00  or 01

but existing schema has 2 column as pk,value(-1,0,1) it will reduce space and easy while maintainance by scaning a single coloumn we can get 3 variable data

Hrusi01 commented:
> @Hrusi01 fix the ci errors

bhaiya apka signoff  missing he

Hrusi01 committed:
update

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
biome fix

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
ci fix

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
ci fix

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
updated the anonymousname

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

neoandmatrix reviewed (commented):
Too big of a PR with many changes at once.
My suggestions :- 
1. Remove all the get route functionality overlapping with #39 .
2. Make changes manageable to review like one pr which adds all the schemas with dto and other one to add the logical and functionality.

CC: @iamanishx @ujsquared

neoandmatrix reviewed (commented):
Changes required.

Hrusi01 committed:
Merge branch 'main' into main

iamanishx commented:
Remove apps/server/src/routers/topics.ts and post.dto,ts from the unrelated to your issue 
@Hrusi01

iamanishx commented:
@Hrusi01 apps/server/src/routers/votes.ts 1st of all why is this under /routes? And again no authentication or authorisation!!

Hrusi01 committed:
sync posts.dto.ts from upstream

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
upstream sync lock posts topic pkgJson

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
new vote schema update

Signed-off-by: Hrusi01 <hrusikeshkar01@gmail.com>

Hrusi01 committed:
updated dto

Hrusi01 commented:
@iamanishx @neoandmatrix 
new schema updated, auth added to vote.ts, added logic separatly in service layer

Hrusi01 commented:
> @Hrusi01 #46 Follow up on that PR AS mentioned that contains all the stuffs from this pr tooo
> 
> Tagging cause you asked last night

but in the commits as i can see there is no overlaping with #46  i had already removed as you told and cleaned the pr 
this pr only has voting schema and the anonymous feature..if there are any other over lap please guide
